convert/main.go

```diff
@@ -84,53 +87,35 @@ func (g *gen) Name() resource.Name {
 }
 
 func (g *gen) Close(ctx context.Context) error {
+	// todo: implement
 	return nil
 }
 
-type MyVec struct {
-	X float64 `json:"x"`
-	Y float64 `json:"y"`
-	Z float64 `json:"z"`
-}
-
-type MyTriangle struct {
-	P0     MyVec `json:"p0"`
-	P1     MyVec `json:"p1"`
-	P2     MyVec `json:"p2"`
-	Normal MyVec `json:"normal"`
-}
-
-func convertPoint(vec r3.Vector) MyVec {
-	return MyVec{vec.X, vec.Y, vec.Z}
-}
-
-func convertTriangle(t *spatialmath.Triangle) MyTriangle {
-	points := t.Points()
-	return MyTriangle{
-		P0:     convertPoint(points[0]),
-		P1:     convertPoint(points[1]),
-		P2:     convertPoint(points[2]),
-		Normal: convertPoint(t.Normal()),
-	}
-}
-
 // DoCommand echos input back to the caller.
 func (g *gen) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
-	mesh, err := g.getMeshFromPC()
+	pcInterface, ok := cmd["pointcloud"]
+	if !ok {
+		return nil, errors.New("command was incorrectly specified")
+	}
+	pc, err := getPointCloudFromBytes(pcInterface)
 	if err != nil {
 		return nil, err
 	}
-	g.logger.Infof("mesh.Triangles(): %v", mesh.Triangles())
-	g.logger.Infof("len(mesh.Triangles()): %v", len(mesh.Triangles()))
-	var asMap []map[string]any
-
-	// v := convertTriangle(mesh)
-	convertible := make([]MyTriangle, 0, len(mesh.Triangles()))
-	for _, t := range mesh.Triangles() {
-		convertible = append(convertible, convertTriangle(t))
+	g.logger.Infof("got the pointcloud from bytes: %v", pc)
+
+	// write to a .pcd file so that mesh reconstruction can read from it
+	err = g.writeToFile(pc)
+	if err != nil {
+		return nil, err
+	}
+	g.logger.Infof("wrote the the pointcloud to a file")
+
+	// get the spatialmath.Mesh
+	mesh, err := g.getMeshFromPointCloud()
+	if err != nil {
+		return nil, err
 	}
-	mapstructure.Decode(convertible, &asMap)
-	println("decoded to", len(asMap))
 
-	return map[string]interface{}{"mesh_triangles": asMap}, nil
+	// generate json representation of mesh
+	return map[string]interface{}{"mesh_triangles": generateMeshJson(mesh)}, nil
```

Comment:
@biotinker I think it makes more sense to just return the .ply file directly over the wire.

This way we can remove `triangles.go`